pkg/filesystem/subvolume.go

```diff
@@ -211,7 +217,73 @@ func Delete(ctx context.Context, clientsets *k8sutil.Clientsets, OperatorNamespa
 	if !check {
 		exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"fs", "subvolume", "rm", fs, subvol, svg, "--retain-snapshots"}, OperatorNamespace, CephClusterNamespace, true)
 		logging.Info("subvolume %q deleted", subvol)
+		deleteOmap(ctx, clientsets, OperatorNamespace, CephClusterNamespace, subvol, fs)
 	} else {
 		logging.Info("subvolume %q is not stale", subvol)
 	}
 }
+
+// deleteOmap deletes omap object and key for the given subvolume.
+func deleteOmap(ctx context.Context, clientsets *k8sutil.Clientsets, OperatorNamespace, CephClusterNamespace, subVol, fs string) {
```

Comment:
what will happen to the omap if the subvolumes are deleted but the omap are not deleted/cleaned up?

Reply:
this is the case when someone deletes the subvolume from the backend. Here we are deleting the stale subvolume and just after that removing it's entry from Omap. 
Here, we are deleting omap of only stale subvolumes, as that's the input we are asking from the user.  In case we  want to cleanup the omap to remove all the stale omaps whose subvolume and pv both are deleted , we need to have separate command, where it will list all the omaps and pv's and verify the stale omap.

Reply:
will open and link one.